R/make-traceability-matrix.R

```diff
@@ -70,6 +70,12 @@ map_functions_to_scripts <- function(exports_df, pkg_source_path, verbose){
   # Search for scripts functions are defined in
   funcs_df <- get_toplevel_assignments(pkg_source_path)
 
+  if(nrow(funcs_df) == 0){
+    # This shouldn't be triggered, and either indicates a bug in `get_toplevel_assignments`,
+    # or an unexpected package setup that we may want to support.
+    warning("No top level assignments were found in the R/ directory. Make sure this was expected.")
+  }
+
   exports_df <- dplyr::left_join(exports_df, funcs_df, by = c("exported_function" = "func"))
```

Comment:
If for some reason the new logic in `get_toplevel_assignments` doesn't return any files, we could still hit the error reported in gh-46:

```
Join columns in `y` must be present in the data.
✖ Problem with `func`.
```

It seems to me that `get_toplevel_assignments` should detect that no files are listed and return early with an empty tibble containing the expected columns.

Reply:
This is somewhat alarming. Might make sense to have Seth test this on his end before merging just to make sure these scenarios were actually addressed. Will look into it more

Reply:
[ pulling in relevant snippet from [outside comment](https://github.com/metrumresearchgroup/mpn.scorecard/pull/47#issuecomment-1736135574) ]

> It doesnt seem to be working for @kyleam ([relevant comment](https://github.com/metrumresearchgroup/mpn.scorecard/pull/47#discussion_r1335039759)) but I cant reproduce it on my end (works as expected).

With this PR, I didn't see a failure with boot and robust.  If I did, I would have said so and not approved this PR.  My point was just that, even if we can't think of another scenario where the file listing would come up empty, we shouldn't have a code path in `get_toplevel_assignments` where it returns a value that will make this spot fail with the same error.

With your update in 45601c067, it no longer does, so we should be good here.